### ISSUE TITLE
GrapheneOS compatibility

### ISSUE THREAD

**Issue by boistordu** [AsteroidOS/AsteroidOSSync#173]
Can't get any file permissions on grapheneOS

**Comment by eLtMosen** [comment]
Thanks for reporting!
Could you clarify on which version of GrapheneOS that happens?

**Comment by domkrm** [comment]
Since version 0.23 the Tarket SDK 33 is used. With this newer target and on newer Android versions the WRITE_EXTERNAL_STORAGE permission has no effect anymore and you can't grant it anymore too:

https://developer.android.com/about/versions/11/privacy/storage

Currently I'm looking how we can fix this (but I have not enough time currently).
We can save Screenshots etc. in den app-specific directory:

https://developer.android.com/reference/android/content/Context#getExternalFilesDirs(java.lang.String)

But I the App is uninstalled this directory will be deleted also.

Second option is to grant access to a specific user-selected directory:

https://developer.android.com/training/data-storage#scoped-storage

**Comment by domkrm** [comment]
I think until we have a solution, we should rollback the `targetSdk to 30` and release a new version. Otherwise you can't install/configure the App, because you can't get over the permission activity, because you get always the error message, that you should grant the permission when giving the storage permission.

**Comment by domkrm** [comment]
The only solution I found to install the newest version of the App is, first install version 0.22, then configure the App and then update the App. This worked for me.

**Comment by jrtberlin** [comment]
> The only solution I found to install the newest version of the App is, first install version 0.22, then configure the App and then update the App. This worked for me.

I'm now on GrapheneOS too and could reproduce your issue. My suspicion is that we implemented scoped storage but didn't remove the legacy permission check.

**Comment by LordPax** [comment]
I have same issue with calyxos

**Comment by jrtberlin** [comment]
> I have same issue with calyxos

Yes. The issue is Android 13 specific and not Graphene specific.

**Comment by jrtberlin** [comment]
Please check if the issue persists with the latest version from F-Droid.

**Comment by LordPax** [comment]
This issue is solved for me

**Comment by smilzo05** [comment]
hello, I have the latest version of graphene os, but I can't get a synchronization with asteroidOs. Are there any solutions today? installing version 22 of asteroidOs didn't solve the problem for me. Thank you

**Comment by eLtMosen** [comment]
@smilzo05 in case you mean 0.22, that would be to expected.
There have been multple versions since that one with explicit fixes for newer Android versions.
Latest version is 0.28 and available on F-Droid
https://f-droid.org/de/packages/org.asteroidos.sync/

**Comment by smilzo05** [comment]
hello, I have the latest version of graphene os, but I can't get a synchronization with asteroidOs. Are there any solutions today? Thank you

**Comment by smilzo05** [comment]
0.28 didn't work so I searched for the solution. in this topic I read that it was solved by installing version 0.22, it was not like that for me

**Comment by AmiiiiKo** [comment]
I have a Pixel 7 with Graphene OS, latest update and a Ticwatch pro 2020, everything seems to work fine except for saving screenshots. I believe it is likely because it hasn't requested any permissions to save the file therefore I cannot configure storage scopes.
I am running the f-droid version, 0.28. I also tried it with Gadgetbridge and that worked as well.
